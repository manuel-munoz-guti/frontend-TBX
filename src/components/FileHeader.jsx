import React from 'react'

export const FileHeader = ({ columns }) => {

  return (
    <thead>
      <tr>
        {
            columns.map( (column) => (
                <th key={column}> {column} </th>
            ))
        }
      </tr>
    </thead>
  )
}
