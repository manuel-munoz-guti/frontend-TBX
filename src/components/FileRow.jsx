import React from 'react'

export const FileRow = ({fileName, line}) => {
  return (
    <tr>
      <td> {fileName} </td>
      <td> {line.text} </td>
      <td> {line.number} </td>
      <td> {line.hex} </td>
    </tr>
  )
}
