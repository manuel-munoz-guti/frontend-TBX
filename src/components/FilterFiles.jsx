import React, { useState } from 'react';
import { Container, Form } from 'react-bootstrap';
import { useDispatch} from 'react-redux';
import { setClearResponse, startLodingFile } from '../store/files/';
import { setNotTyping, setTyping } from '../store/ui/uiSlice';

export const FilterFiles = () => {
  const [filter, setFilter] = useState('');
  const dispatch = useDispatch();

  const onChangeValue = ({target}) => {
    setFilter(target.value);
    dispatch( setTyping() );
    dispatch( setClearResponse() );
  }

  const onSubmitForm = (event) => {
    event.preventDefault();
    dispatch( setNotTyping() );
    if(!filter) return ;
    dispatch( startLodingFile(filter) );
    setFilter('');
  }

  return (
    <Container className="p-3">
      <h6 className="header animate__animated animate__backInLeft" style={{ color: 'white'}}>File Filter:</h6>
      <Form onSubmit={onSubmitForm}>
        <Form.Control 
          size="sm" 
          type="text" 
          placeholder="Enter the file name to search" 
          className='animate__animated animate__backInLeft' 
          style={{ width: '30%' }}
          name='filter'
          value={ filter }
          onChange={ onChangeValue }
          />
      </Form>
    </Container>
  )
}
