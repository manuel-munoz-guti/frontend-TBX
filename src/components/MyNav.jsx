import React from 'react';
import Container from 'react-bootstrap/Container';
import Navbar from 'react-bootstrap/Navbar';

export const MyNav = () => {
    return (
        <>
            <Navbar bg="dark" variant="dark">
                <Container>
                    <Navbar.Brand href="#home">
                    <img
                        alt=""
                        src="../../src/favicon.svg"
                        width="30"
                        height="30"
                        className="d-inline-block align-top"
                    />{'  '}
                    REACT CHALLENGE - TBX
                    </Navbar.Brand>
                </Container>
            </Navbar>
        </>
    )
}
