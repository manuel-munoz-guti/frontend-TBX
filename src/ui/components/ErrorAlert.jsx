import React from 'react'
import { Alert } from 'react-bootstrap'

export const ErrorAlert = ({ message , code }) => {
  return (
    <Alert variant='danger' className='animate__animated animate__flash'>
          Opps! something went wrong, contact the Administrator. CODE {code}: ({message}  )
    </Alert>
  )
}
