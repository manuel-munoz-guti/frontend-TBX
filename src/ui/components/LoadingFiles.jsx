import React, { useEffect, useState } from 'react'
import { Container, ProgressBar } from 'react-bootstrap'

export const LoadingFiles = () => {

  const [now, setNow] = useState(0);
  
  useEffect(() => {
    for(let i=0; i<4; i++){
      setTimeout( () => {
        setNow( prev => prev + 25);
      }, 300);
    }
  }, [])
  

  return (
    <div>
      <Container className='my-5 py-5'>

        <ProgressBar animated now={now} label={`${now}%`} visuallyHidden />
        Loading({`${now/2}%`})...

      </Container>
    </div>
  )
}
