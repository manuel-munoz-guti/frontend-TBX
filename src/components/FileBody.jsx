import React from 'react'
import { FileRow } from './FileRow'

export const FileBody = ({data}) => {
  return (
    <tbody>
        {
            data.map( (element) => (
               element.lines.map( (line) => (
                 <FileRow key={line.hex} line={line} fileName={element.file}/>
               ))
            ))
        }
    </tbody>
  )
}
