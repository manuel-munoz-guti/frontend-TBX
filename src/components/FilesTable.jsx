import React, { useState } from 'react';
import { Table } from 'react-bootstrap';
import { useSelector } from 'react-redux';
import { FileBody } from './FileBody';
import { FileHeader } from './FileHeader';

export const FilesTable = () => {
  const [header, setHeader] = useState(['File Name', 'Text', 'Number', 'Hex']);   // header's columns for table
  const { data } = useSelector( state => state.files );

  return (
    <div className='animate__animated animate__backInUp'>
      <Table striped bordered hover responsive="sm">
        
        <FileHeader columns={header} />
        
        <FileBody data={data}/>
      
      </Table>
    </div>
  )
}
