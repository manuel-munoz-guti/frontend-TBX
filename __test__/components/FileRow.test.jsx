const { render } = require("@testing-library/react");
import { Table } from 'react-bootstrap';
import { FileRow } from '../../src/components/FileRow';

describe('Test On component FileRow', () => {

    const file = 'test1.csv';
    const line = {
        "text": "UFNWbIeTSjoySbvQBMbLqp",
        "number": 87,
        "hex": "7ac6f24ef988372576935d449d02176f"
    };
    

    test('Should match with snapshot', () => {
        const {container} = render( 
            <Table>
        
            <tbody>
                <FileRow key={line.hex} line={line} fileName={file}/>
            </tbody>
          
            </Table>
     
        );
        expect(container).toMatchSnapshot();
    });

    
});