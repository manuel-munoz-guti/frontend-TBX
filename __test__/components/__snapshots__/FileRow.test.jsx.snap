// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test On component FileRow Should match with snapshot 1`] = `
<div>
  <table
    class="table"
  >
    <tbody>
      <tr>
        <td>
           
          test1.csv
           
        </td>
        <td>
           
          UFNWbIeTSjoySbvQBMbLqp
           
        </td>
        <td>
           
          87
           
        </td>
        <td>
           
          7ac6f24ef988372576935d449d02176f
           
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
